System/Process.hs

```diff
@@ -857,6 +862,19 @@ terminateProcess ph = do
 -- ----------------------------------------------------------------------------
 -- Interface to C bits
 
+#if defined(__wasm__)
+
+c_terminateProcess :: PHANDLE -> IO CInt
+c_terminateProcess _ = ioError (ioeSetLocation unsupportedOperation "terminateProcess")
```

Comment:
Yes, this is better than the previous iteration. I probably would have factored out this pattern into a `unsupportedByWasm` helper function living in `System.Process.Internal` but I'm fine with this as well.